src/js/view/components/LogViewer.js

```diff
@@ -129,63 +166,57 @@ const LogViewer = props => {
     props.totalNrOfFilteredLines[props.source.path]
   ]);
 
+  useEffect(() => {
+    if (filterInput.length !== 0) {
+      fetch();
+    } else if (filterInput.length === 0) {
+      if (props.totalNrOfFilteredLines[props.source.path] !== 0) {
+        clearFilteredLines(props.dispatch, props.source.path);
+      }
+    }
+  }, [filterInput, props.logs[props.source.path]]);
+
   useEffect(() => {
     /* Effect for when a new filter or highlight is applied,
     send the lines to be filtered and highlighted again */
-    let logs;
     filterRef.current = filterInput;
-
     if (filterInput.length !== 0) {
-      fetchFilteredLinesFromBackend(
-        props.source.path,
-        parseRegExp(filterInput).toString()
-      );
-      logs = props.filteredLogs[props.source.path];
-    } else {
-      logs = props.logs[props.source.path];
-    }
-
-    if (logs) {
-      // Reset the previous lines count, as all lines should be wiped.
-      previousFilteredLinesLength.current = 0;
-      previousLinesLength.current = 0;
+      // let newLines = filteredLogs.slice(previousFilteredLinesLength.current);
       sendMessageToHiddenWindow({
-        logs
+        // sendLinesOneByOne: previousFilteredLinesLength.current > 0 ? true : false,
```

Comment:
I am not sure if leaving these lines commented will make sense for future developers. We understand it when reading the pr and the bug ticket. Maybe remove the comments and save them as a note in the bug ticket you wrote?  Or ad an explanatory comment? Not sure atm what would be the best solution. I might need to read the explanation again and come back.

Reply:
Yes, makes sense to at least leave an explanatory comment together with the code to hint why this is still here and refer it to ticket #543 .

Reply:
Other than that everything seems to work well on my machine. I have not been able to recreate the bug and the filtering process does not cost as much cpu as before. Thumbs up!

Reply:
Good idea! I added explanatory comments that I hope will make it clearer and refered to #543